libr/bin/mangling/swift-sd.c

```diff
@@ -262,34 +277,41 @@ R_API char *r_bin_demangle_swift(const char *s, bool syscmd, bool trylib) {
 		}
 	}
 	if (tail) {
-		if (*p) {
-			p++;
-		}
+		p = str_seek (p, 1);
 	} else {
-		if (*p && p[1]) {
-			p += 2;
-		}
+		p = str_seek (p, 2);
```

Comment:
if (*p && p[1]) is not covered in this case.
```suggestion
		if (*p && p[1]) {
			p = str_seek (p, 2);
		}
```

Reply:
thanks